marklogic-data-hub/src/main/java/com/marklogic/hub/impl/VersionInfo.java

```diff
@@ -0,0 +1,55 @@
+package com.marklogic.hub.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.hub.HubClient;
+import com.marklogic.hub.dataservices.SystemService;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * This is intended to replace some of the Versions class with a HubClient-friendly and non-Spring-dependent way of
+ * accessing version data.
+ */
+public class VersionInfo {
+
+    private String hubVersion;
+    private String markLogicVersion;
+
+    public static VersionInfo newVersionInfo(HubClient hubClient) {
+        JsonNode json = SystemService.on(hubClient.getStagingClient()).getVersions();
+        return new VersionInfo(
+            json.get("hubVersion").asText(),
+            json.get("markLogicVersion").asText()
+        );
+    }
+
+    private VersionInfo(String hubVersion, String markLogicVersion) {
+        this.hubVersion = hubVersion;
+        this.markLogicVersion = markLogicVersion;
+    }
+
+    public String getHubVersion() {
+        return hubVersion;
+    }
+
+    public String getMarkLogicVersion() {
+        return markLogicVersion;
+    }
+
+    /**
+     * @return the version of the build containing this class
+     */
+    public static String getBuildVersion() {
+        Properties properties = new Properties();
+        try (InputStream inputStream = VersionInfo.class.getClassLoader().getResourceAsStream("version.properties")) {
+            properties.load(inputStream);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to get library version from version.properties, cause: " + e.getMessage(), e);
+        }
+
+        String version = properties.getProperty("version");
+        return "${project.version}".equals(version) ? "5.3-SNAPSHOT" : version;
```

Comment:
Can we add this file to the `updateVersion` task that's in root build.gradle?
The `updateVersion` task keeps track of all the hardcoded version locations.

Reply:
Good idea, done